toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -244,6 +244,9 @@ func CreatePartitions(diskDevPath string, disk configuration.Disk, rootEncryptio
 
 	// Create new partition table
 	partitionTableType := disk.PartitionTableType
+	if partitionTableType == "mbr" {
```

Comment:
We already have a few methods defined for the `PartitionTableType` type. I'd suggest adding another one like `ForParted` or `AsPartedArg` to keep all code related to operating with this type in one place.

Reply:
That makes sense. I'll refactor and resubmit.